crates/router/src/connector/iatapay/transformers.rs

```diff
@@ -139,7 +150,14 @@ impl TryFrom<&types::ConnectorAuthType> for IatapayAuthType {
                 merchant_id: key1.to_owned(),
                 client_secret: api_secret.to_owned(),
             }),
-            _ => Err(errors::ConnectorError::FailedToObtainAuthType)?,
+
+            types::ConnectorAuthType::HeaderKey { .. }
+            | types::ConnectorAuthType::BodyKey { .. }
+            | types::ConnectorAuthType::MultiAuthKey { .. }
+            | types::ConnectorAuthType::CurrencyAuthKey { .. }
+            | types::ConnectorAuthType::NoKey { .. } => {
+                Err(errors::ConnectorError::FailedToObtainAuthType)?
+            }
```

Comment:
Please revert back this change as handling wild card match arms for AuthType is not in the scope. Rest looks good to me!

Reply:
I have reverted back the unwanted changes